README.md

```diff
@@ -366,4 +366,136 @@ func (s *snapshot) DeserializeSnapshot(m eventsourcing.DeserializeFunc, b []byte
 	s.unexported = snap.UnExported
 	return nil
 }
+```
+
+## Projections - Work in progress
+
+Introduction
+
+### ProjectionHandler
+
+The Projection handler is the central part where projections are created. It's available from the event repository by the `eventrepo.Projections()` property but can also be created by itself.
+
+```go
+// access via the event repositort
+eventRepo := eventsourcing.NewEventRepository(eventstore)
+ph := eventRepo.Projections
+
+// creates without the event repository
+ph := eventsourcing.NewProjectionHandler(register, encoder)
+```
+
+The projection handler include the event register and a encoder to deserialize events from an event store to an application event.
+
+### Projection
+
+A projection is created from the projection handler via the `Projection()` method. The method takes a `fetchFunc` and a `callbackFunc` and return a pointer to a Projection.
+
+```go
+p := ph.Projection(f fetchFunc, c callbackFunc)
+```
+
+The fetchFunc must return a `core.Iterator` and a `error`. This is the same signature that event stores return when they return events.
+
+```go
+type fetchFunc func() (core.Iterator, error)
+```
+
+The `callbackFunc` is called for every iterated event. The event is typed and can be handled in the same way as the aggregate `Transition()` method.
+
+```go
+type callbackFunc func(e eventsourcing.Event) error
+```
+
+Example: Creates a projection that fetch all events from an event store and handle them in the callbackF.
+
+```go
+p := eventRepo.Projections.Projection(es.All(0, 1), func(event eventsourcing.Event) error {
```

Comment:
Seeing this code example makes me want an alias for this on the eventRepo:

```go
p := eventRepo.Project(es.All(0, 1), func(event eventsourcing.Event) error {
// ...
})

But that's just nice-to-have, nothing crucial.
```

Reply:
By alias you mean a method on the event repo directly to create a projection?

Reply:
Yes, something like that. But that could be added later if needed!